java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java

```diff
@@ -75,18 +75,19 @@ public StringHashTableDictionary(int initialCapacity, float loadFactor) {
     this.capacity = initialCapacity;
     this.loadFactor = loadFactor;
     this.keyOffsets = new DynamicIntArray(initialCapacity);
-    initHashBuckets(initialCapacity);
+    initHashBuckets(false);
     this.threshold = (int)Math.min(initialCapacity * loadFactor, MAX_ARRAY_SIZE + 1);
   }
 
-  private void initHashBuckets(int capacity) {
-    DynamicIntArray[] buckets = new DynamicIntArray[capacity];
-    for (int i = 0; i < capacity; i++) {
+  private void initHashBuckets(final boolean reinit) {
```

Comment:
I would even remove the parameter of this method -- we just need to ensure that hashBuckets is initialized right?
So we can just check if its non-null (assuming the capacity remains the same) otherwise create the new buckets

Reply:
Ha. Yes. You're absolutely correct.  I will change this to simplify.